### ISSUE TITLE
conda-pack fails for this environment spec

### ISSUE THREAD

**Issue by bharathguvvala** [conda/conda-pack#63]
Got the following exception for the spec given below even though none of the packages listed in the exception (jinja2) are installed/overwritten by pip at all.
```
Collecting packages...
CondaPackError:
Files managed by conda were found to have been deleted/overwritten in the
following packages:

- jinja2=u'2.10'

This is usually due to `pip` uninstalling or clobbering conda managed files,
resulting in an inconsistent environment. Please check your environment for
conda/pip conflicts using `conda list`, and fix the environment by ensuring
only one version of each package is installed (conda preferred).
```

Environment config:
```
channels:
  - conda-forge
  - defaults
dependencies:
  - attrs=18.2.0=py_0
  - blas=1.1=openblas
  - ca-certificates=2018.10.15=ha4d7672_0
  - certifi=2018.10.15=py27_1000
  - click=7.0=py_0
  - cloudpickle=0.2.1=py27_2
  - conda-pack=0.3.0=py_0
  - docopt=0.6.2=py_1
  - fastavro=0.21.13=py27h470a237_0
  - filechunkio=1.8=py_2
  - flask=1.0.2=py_2
  - gunicorn=19.9.0=py27_1000
  - ipykernel=4.10.0=py27_1
  - ipython=5.8.0=py27_0
  - itsdangerous=1.1.0=py_0
  - jinja2=2.10=py_1
  - jsonschema=3.0.0a3=py27_1000
  - jupyter_client=5.2.3=py_1
  - jupyter_core=4.4.0=py_0
  - krb5=1.16.2=hbb41f41_0
  - libgfortran=3.0.0=1
  - libsodium=1.0.16=h470a237_1
  - libxgboost=0.80=hfc679d8_1
  - markupsafe=1.1.0=py27h470a237_0
  - mkl_fft=1.0.6=py27_0
  - mkl_random=1.0.2=py27_0
  - mock=2.0.0=py27_1000
  - numpy=1.13.3=py27_blas_openblas_201
  - openblas=0.2.20=8
  - openssl=1.0.2p=h470a237_1
  - pandas=0.23.4=py27hf8a1672_0
  - pip=10.0.1=py27_0
  - psutil=5.4.8=py27h470a237_0
  - py-xgboost=0.80=py27hfc679d8_1
  - pyrsistent=0.14.5=py27h470a237_1
  - python=2.7.15=h33da82c_4
  - python-hdfs=2.1.0=py_1
  - pyyaml=3.13=py27h470a237_1
  - pyzmq=17.1.2=py27hae99301_1
  - requests=2.20.1=py27_1000
  - retrying=1.3.3=py_2
  - scikit-learn=0.19.1=py27_blas_openblas_201
  - scipy=1.1.0=py27_blas_openblash7943236_201
  - simplejson=3.16.1=py27h470a237_0
  - sqlalchemy=1.2.14=py27h470a237_0
  - thrift=0.11.0=py27hfc679d8_1
  - xgboost=0.80=py27hfc679d8_1
  - zeromq=4.2.5=hfc679d8_6
  - _py-xgboost-mutex=2.0=cpu_0
  - asn1crypto=0.24.0=py27_0
  - backports=1.0=py27_1
  - backports.functools_lru_cache=1.5=py27_1
  - backports.shutil_get_terminal_size=1.0.0=py27_2
  - backports.weakref=1.0rc1=py27_0
  - backports_abc=0.5=py27_0
  - bleach=1.5.0=py27_0
  - boto=2.49.0=py27_0
  - boto3=1.9.35=py27_0
  - botocore=1.12.35=py27_0
  - bz2file=0.98=py27_1
  - cffi=1.11.5=py27he75722e_1
  - chardet=3.0.4=py27_1
  - cryptography=2.3.1=py27hc365091_0
  - cycler=0.10.0=py27hc7354d3_0
  - dbus=1.13.2=h714fa37_1
  - decorator=4.3.0=py27_0
  - docutils=0.14=py27_0
  - enum34=1.1.6=py27_1
  - expat=2.2.6=he6710b0_0
  - fontconfig=2.13.0=h9420a91_0
  - freetype=2.9.1=h8a8886c_1
  - funcsigs=1.0.2=py27h83f16ab_0
  - functools32=3.2.3.2=py27_1
  - futures=3.2.0=py27_0
  - gensim=1.0.1=np113py27_0
  - glib=2.56.2=hd408876_0
  - gst-plugins-base=1.14.0=hbbd80ab_1
  - gstreamer=1.14.0=hb453b48_1
  - h5py=2.8.0=py27h989c5e5_3
  - hdf5=1.10.2=hba1933b_1
  - html5lib=0.9999999=py27_0
  - icu=58.2=h9c2bf20_1
  - idna=2.7=py27_0
  - intel-openmp=2019.0=118
  - ipaddress=1.0.22=py27_0
  - ipython_genutils=0.2.0=py27h89fb69b_0
  - jmespath=0.9.3=py27_0
  - jpeg=9b=h024ee3a_2
  - keras=2.1.3=py27_0
  - kiwisolver=1.0.1=py27hf484d3e_0
  - libedit=3.1.20170329=h6b74fdf_2
  - libffi=3.2.1=hd88cf55_4
  - libgcc-ng=8.2.0=hdf63c60_1
  - libgfortran-ng=7.3.0=hdf63c60_0
  - libpng=1.6.35=hbc83047_0
  - libprotobuf=3.6.1=hd408876_0
  - libstdcxx-ng=8.2.0=hdf63c60_1
  - libuuid=1.0.3=h1bed415_2
  - libxcb=1.13=h1bed415_1
  - libxml2=2.9.8=h26e45fe_1
  - linecache2=1.0.0=py27_0
  - markdown=3.0.1=py27_0
  - matplotlib=2.2.2=py27hb69df0a_2
  - mkl=2019.0=118
  - ncurses=6.1=hf484d3e_0
  - pathlib2=2.3.2=py27_0
  - patsy=0.5.1=py27_0
  - pbr=5.1.0=py27_0
  - pcre=8.42=h439df22_0
  - pexpect=4.6.0=py27_0
  - pickleshare=0.7.5=py27_0
  - prompt_toolkit=1.0.15=py27_0
  - protobuf=3.6.1=py27he6710b0_0
  - ptyprocess=0.6.0=py27_0
  - pycparser=2.19=py27_0
  - pygments=2.2.0=py27h4a8b6f5_0
  - pykerberos=1.1.14=py27h2de3caf_2
  - pyopenssl=18.0.0=py27_0
  - pyparsing=2.3.0=py27_0
  - pyqt=5.9.2=py27h22d08a2_1
  - pysocks=1.6.8=py27_0
  - python-dateutil=2.7.5=py27_0
  - pytz=2018.7=py27_0
  - qt=5.9.6=h8703b6f_2
  - readline=7.0=h7b6447c_5
  - requests-kerberos=0.12.0=py27_0
  - s3transfer=0.1.13=py27_0
  - scandir=1.9.0=py27h14c3975_0
  - seaborn=0.8.1=py27_0
  - setuptools=40.5.0=py27_0
  - simplegeneric=0.8.1=py27_2
  - singledispatch=3.4.0.3=py27h9bcb476_0
  - sip=4.19.13=py27he6710b0_0
  - six=1.11.0=py27_1
  - smart_open=1.7.1=py27_0
  - sqlite=3.25.2=h7b6447c_0
  - statsmodels=0.9.0=py27h035aef0_0
  - subprocess32=3.5.3=py27h7b6447c_0
  - tensorflow=1.3.0=0
  - tensorflow-base=1.3.0=py27h0dbb4d0_1
  - tensorflow-tensorboard=1.5.1=py27hf484d3e_1
  - termcolor=1.1.0=py27_1
  - tk=8.6.8=hbc83047_0
  - tornado=5.1.1=py27h7b6447c_0
  - tqdm=4.26.0=py27h28b3542_0
  - traceback2=1.4.0=py27_0
  - traitlets=4.3.2=py27_0
  - unittest2=1.1.0=py27_0
  - urllib3=1.23=py27_0
  - wcwidth=0.1.7=py27h9e3e1ab_0
  - werkzeug=0.14.1=py27_0
  - wheel=0.32.2=py27_0
  - xz=5.2.4=h14c3975_4
  - yaml=0.1.7=had09818_2
  - zlib=1.2.11=ha838bed_2
  - pip:
    - hdfs==2.1.0
    - state-machine==0.2.10
```

**Comment by bharathguvvala** [comment]
FYI. ```- flask=1.0.2=py_2``` depends on ```- jinja2=2.10=py_1```

**Comment by bharathguvvala** [comment]
This was using conda-pack=0.2.1

**Comment by jcrist** [comment]
Thanks for the issue, I'm able to reproduce it with the following:

```
conda create -n test-pack python=2.7.15 jinja2=2.10 -y -c conda-forge -c defaults
```

Debugging now.

**Comment by bharathguvvala** [comment]
I just tried this with the latest conda-pack=0.3.0 and this error doesn't occur in that case.

**Comment by jcrist** [comment]
Hmmm, I tried it with conda-pack 0.3.0, and still got the error.

**Comment by bharathguvvala** [comment]
> Hmmm, I tried it with conda-pack 0.3.0, and still got the error.

aah, okay.

**Comment by jcrist** [comment]
Should be fixed by #64.

**Comment by jcrist** [comment]
This is fixed in master, I'll push a bugfix release in the next couple of days.

**Comment by bharathguvvala** [comment]
@jcrist Has a release been made with these changes?

**Comment by jcrist** [comment]
Apologies, this slipped my mind. I just released version 0.3.1, on PyPI and conda-forge now.

**Comment by Woodstonelee** [comment]
I'm using conda pack v0.3.1 but it looks like I have come across this problem again. None of the package listed, mpich2, is installed or overwritten by pip, but conda pack complains. Once I removed mpich2 from the environment, the conda pack can succeed. 

```
Collecting packages...
CondaPackError: 
Files managed by conda were found to have been deleted/overwritten in the
following packages:

- mpich2='1.4.1p1'

This is usually due to `pip` uninstalling or clobbering conda managed files,
resulting in an inconsistent environment. Please check your environment for
conda/pip conflicts using `conda list`, and fix the environment by ensuring
only one version of each package is installed (conda preferred).
```

And the environment config

```
channels:
  - defaults
dependencies:
  - asn1crypto=0.24.0=py36_0
  - backcall=0.1.0=py36_0
  - blas=1.0=mkl
  - bleach=3.0.2=py36_0
  - bzip2=1.0.6=h14c3975_5
  - ca-certificates=2019.1.23=0
  - cairo=1.14.12=h8948797_3
  - cartopy=0.17.0=py36hbb7e04d_1
  - certifi=2019.3.9=py36_0
  - cffi=1.11.5=py36he75722e_1
  - chardet=3.0.4=py36_1
  - cryptography=2.4.2=py36h1ba5d50_0
  - curl=7.63.0=hbc83047_1000
  - cycler=0.10.0=py36_0
  - dbus=1.13.2=h714fa37_1
  - decorator=4.3.0=py36_0
  - entrypoints=0.2.3=py36_2
  - expat=2.2.6=he6710b0_0
  - fontconfig=2.13.0=h9420a91_0
  - freetype=2.9.1=h8a8886c_1
  - freexl=1.0.5=h14c3975_0
  - gdal=2.3.3=py36hbb2a789_0
  - geos=3.7.1=he6710b0_0
  - giflib=5.1.4=h14c3975_1
  - glib=2.56.2=hd408876_0
  - gmp=6.1.2=h6c8ec71_1
  - gst-plugins-base=1.14.0=hbbd80ab_1
  - gstreamer=1.14.0=hb453b48_1
  - hdf4=4.2.13=h3ca952b_2
  - hdf5=1.10.4=hb1b8bf9_0
  - icu=58.2=h9c2bf20_1
  - idna=2.8=py36_0
  - intel-openmp=2019.1=144
  - ipykernel=5.1.0=py36h39e3cac_0
  - ipython=7.2.0=py36h39e3cac_0
  - ipython_genutils=0.2.0=py36_0
  - jedi=0.13.1=py36_0
  - jinja2=2.10=py36_0
  - joblib=0.13.2=py36_0
  - jpeg=9b=h024ee3a_2
  - json-c=0.13.1=h1bed415_0
  - jsonschema=2.6.0=py36_0
  - jupyter_client=5.2.3=py36_0
  - jupyter_core=4.4.0=py36_0
  - jupyterlab=0.35.3=py36_0
  - jupyterlab_server=0.2.0=py36_0
  - kealib=1.4.7=hd0c454d_6
  - kiwisolver=1.0.1=py36hf484d3e_0
  - krb5=1.16.1=h173b8e3_7
  - libboost=1.67.0=h46d08c1_4
  - libcurl=7.63.0=h20c2e04_1000
  - libdap4=3.19.1=h6ec2957_0
  - libedit=3.1.20170329=h6b74fdf_2
  - libffi=3.2.1=hd88cf55_4
  - libgcc-ng=8.2.0=hdf63c60_1
  - libgdal=2.3.3=h2e7e64b_0
  - libgfortran-ng=7.3.0=hdf63c60_0
  - libkml=1.3.0=h590aaf7_4
  - libnetcdf=4.6.1=h11d0813_2
  - libpng=1.6.35=hbc83047_0
  - libpq=11.1=h20c2e04_0
  - libsodium=1.0.16=h1bed415_0
  - libspatialite=4.3.0a=hb08deb6_19
  - libssh2=1.8.0=h1ba5d50_4
  - libstdcxx-ng=8.2.0=hdf63c60_1
  - libtiff=4.0.9=he85c1e1_2
  - libuuid=1.0.3=h1bed415_2
  - libxcb=1.13=h1bed415_1
  - libxml2=2.9.8=h26e45fe_1
  - libxslt=1.1.33=h7d1a2b0_0
  - llvmlite=0.28.0=py36hd408876_0
  - lxml=4.3.0=py36hefd8a0e_0
  - markupsafe=1.1.0=py36h7b6447c_0
  - matplotlib=3.0.2=py36h5429711_0
  - mistune=0.8.4=py36h7b6447c_0
  - mkl=2018.0.3=1
  - mkl_fft=1.0.6=py36h7dd41cf_0
  - mkl_random=1.0.1=py36h4414c95_1
  - mpi4py=2.0.0=py36_2
  - mpich2=1.4.1p1=0
  - nbconvert=5.3.1=py36_0
  - nbformat=4.4.0=py36_0
  - ncurses=6.1=he6710b0_1
  - notebook=5.7.2=py36_0
  - numba=0.43.1=py36h962f231_0
  - numpy=1.15.4=py36h1d66e8a_0
  - numpy-base=1.15.4=py36h81de0dd_0
  - olefile=0.46=py36_0
  - openjpeg=2.3.0=h05c96fa_1
  - openssl=1.1.1b=h7b6447c_1
  - owslib=0.17.0=py36_0
  - pandas=0.23.4=py36h04863e7_0
  - pandoc=2.2.3.2=0
  - pandocfilters=1.4.2=py36_1
  - parso=0.3.1=py36_0
  - pcre=8.42=h439df22_0
  - pexpect=4.6.0=py36_0
  - pickleshare=0.7.5=py36_0
  - pillow=5.4.1=py36h34e0f95_0
  - pip=19.0.1=py36_0
  - pixman=0.34.0=hceecf20_3
  - poppler=0.65.0=h581218d_1
  - poppler-data=0.4.9=0
  - proj4=5.2.0=he6710b0_1
  - prometheus_client=0.4.2=py36_0
  - prompt_toolkit=2.0.7=py36_0
  - ptyprocess=0.6.0=py36_0
  - pycparser=2.19=py36_0
  - pyepsg=0.4.0=py36_0
  - pygments=2.2.0=py36_0
  - pykdtree=1.3.1=py36hdd07704_2
  - pyopenssl=18.0.0=py36_0
  - pyparsing=2.3.0=py36_0
  - pyproj=1.9.6=py36h14380d9_0
  - pyqt=5.9.2=py36h05f1152_2
  - pyshp=2.0.1=py36_0
  - pysocks=1.6.8=py36_0
  - python=3.6.7=h0371630_0
  - python-dateutil=2.7.5=py36_0
  - pytz=2018.7=py36_0
  - pyzmq=17.1.2=py36h14c3975_0
  - qt=5.9.7=h5867ecd_1
  - readline=7.0=h7b6447c_5
  - requests=2.21.0=py36_0
  - scikit-learn=0.20.1=py36h4989274_0
  - scipy=1.1.0=py36hfa4b5c9_1
  - send2trash=1.5.0=py36_0
  - setuptools=40.6.2=py36_0
  - shapely=1.6.4=py36h86c5351_0
  - sip=4.19.8=py36hf484d3e_0
  - six=1.11.0=py36_1
  - sqlalchemy=1.2.14=py36h7b6447c_0
  - sqlite=3.26.0=h7b6447c_0
  - terminado=0.8.1=py36_1
  - testpath=0.4.2=py36_0
  - tk=8.6.8=hbc83047_0
  - tornado=5.1.1=py36h7b6447c_0
  - traitlets=4.3.2=py36_0
  - urllib3=1.24.1=py36_0
  - wcwidth=0.1.7=py36_0
  - webencodings=0.5.1=py36_1
  - wheel=0.32.3=py36_0
  - xerces-c=3.2.2=h780794e_0
  - xlrd=1.1.0=py36_1
  - xz=5.2.4=h14c3975_4
  - zeromq=4.2.5=hf484d3e_1
  - zlib=1.2.11=h7b6447c_3
  - pip:
    - attrs==18.2.0
    - floweaver==2.0.0a4
    - ipysankeywidget==0.2.5
    - ipywidgets==7.4.2
    - networkx==1.11
    - palettable==3.1.1
    - widgetsnbextension==3.4.2
```
